clusterMembership/dynamicCluster.go

```diff
@@ -14,13 +11,13 @@ type dynamicCluster struct {
 /*
  * Returns a Snapshot of the Cluster Membership state
  */
-func (c dynamicCluster) Members() []string {
+func (c dynamicCluster) Members() []Node {
```

Comment:
Hrm, it might be worth changing these from (c dynamicCluster) to (c *dynamicCluster). The former can mean it has to make a copy of the struct for each method call. As a result, most code seems to be in terms of pointers, not value types.